src/t8_geometry/t8_geometry_implementations/t8_geometry_occ.hxx

```diff
@@ -250,16 +251,73 @@ public:
    *  The edge has to lie on the face.
    * \param [in]  edge_index     The index of the occ edge.
    * \param [in]  face_index     The index of the occ face.
+   * \param [in]  num_face_nodes The number of the face nodes.
    * \param [in]  edge_param     The parameter on the edge.
+   * \param [in]  surface_param  The parameters of the surface nodes.
+   *                             When provided, there are additional checks for closed geometries.
+   *                             If there are no surface parameter
+   *                             to pass in to the function, you can pass NULL.
    * \param [out] face_params    The corresponding parameters on the face.
    */   
   void
   t8_geom_edge_parameter_to_face_parameters(const int edge_index, 
-                                            const int face_index, 
+                                            const int face_index,
+                                            const int num_face_nodes,
                                             const double edge_param, 
+                                            const double* surface_params,
                                             double* face_params) const;
+  
+  /** Finds the parametric bounds of a occ surface.
+   * \param [in]  surface_index   The index of the occ face.
+   * \param [out] bounds          The parametric bounds of the occ surface.
+   */   
+  void
+  t8_geom_get_surface_parametric_bounds(const int surface_index,
+                                        double *bounds) const;
+
+  /** Finds the parametric bounds of an occ edge.
+   * \param [in]  edge_index   The index of the occ edge.
+   * \param [out] bounds       The parametric bounds of the occ edge.
+   */  
+  void
+  t8_geom_get_edge_parametric_bounds(const int edge_index,
+                                     double *bounds) const;
```

Comment:
You have to decide if you use face/edge ode surface/curve. Please do not mix them up